packages/@lwc/synthetic-shadow/src/3rdparty/polymer/text-content.ts

```diff
@@ -17,15 +17,17 @@ subject to an additional IP rights grant found at http://polymer.github.io/PATEN
 // This code is inspired by Polymer ShadyDOM Polyfill
 
 import { getFilteredChildNodes } from '../../faux-shadow/traverse';
-import { ELEMENT_NODE } from '../../env/node';
+import { ELEMENT_NODE, COMMENT_NODE } from '../../env/node';
 
 export function getTextContent(node: Node): string {
     switch (node.nodeType) {
         case ELEMENT_NODE: {
             const childNodes = getFilteredChildNodes(node);
             let content = '';
             for (let i = 0, len = childNodes.length; i < len; i += 1) {
-                content += getTextContent(childNodes[i]);
+                if (childNodes[i].nodeType !== COMMENT_NODE) {
```

Comment:
Should we be more conservative here and implement a whitelist for `Node.TEXT_NODE`?

Reply:
it is not about text node only... elements and eveyething that can have a text inside is valid here.

Reply:
another optimization here is to cache the `childNodes[i]` in a const, instead of accessing it twice in the same block.